packages/subiquity_client/lib/src/server/process.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:io';
+import 'package:path/path.dart' as p;
+
+import 'common.dart';
+import 'pidfile.dart';
+
+// Prefer local curtin and probert python modules that are pinned to the
+// correct versions
+Map<String, String> _pythonPath(String subiquityPath) {
+  final pythonPath = (Platform.environment['PYTHONPATH'] ?? '').split(':');
+  pythonPath.add(subiquityPath);
+  pythonPath.add(p.join(subiquityPath, 'curtin'));
+  pythonPath.add(p.join(subiquityPath, 'probert'));
+  return {'PYTHONPATH': pythonPath.join(':')};
+}
+
+/// Manages the Subiquity process.
+/// This class knows the many ways with which we may want to start Subiquity.
+/// Starting the server builds on top of [Process.start], so this launcher
+/// can be customized by implementing factory constructors that adjust the
+/// [command] and [args] (and possibly some factories to enable enhancing the
+/// command line arguments, working directory and environment with the results
+/// of asynchronous computations).
+/// The example below would allow starting subiquity inside a WSL2 distro
+/// while the GUI client is running on the Windows host.
+///
+/// ```dart
+/// /// Able to start the server inside [distroName] on WSL from the Windows host.
+/// factory SubiquityProcess.wsl(String distroName, String command,
+///     {List<String>? args,
+///     Future<void>? defer,
+///     Future<void> Function()? onProcessStart,
+///     Future<List<String>> Function()? argumentsFactory}) {
```

Comment:
was this meant to be a map of environment vars?

Reply:
Leftovers as well. Should have been just removed.